lib/taxman/taxman2023/qc_i.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Taxman2023
+  # This calculates the total annual taxable income from the income for the period for quebec
+  class QcI < Factor
+    def self.params
+      %i[p qc_g f qc_h qc_csa qc_j qc_j1]
+    end
+    attr_reader *params
+
+    def amount
+      i = p * (qc_g - f - qc_h - qc_csa) - qc_j - qc_j1
+      (i < 0) ? 0 : i
```

Comment:
```suggestion
      [p * (qc_g - f - qc_h - qc_csa) - qc_j - qc_j1, 0].max
```

This keeps it consistent with the rest of the codebase.

Reply:
you know i had it like that, but the rspec failed weirdly in way that did not happen in the console

Reply:
weird - and converting 0 to BigDecimal does not help... 
<img width="855" alt="image" src="https://github.com/Humi-HR/taxman/assets/134647356/3e37fd6c-bb2b-4343-b293-0c0dd70c9ee6">

Reply:
```
      [p * (qc_g - f - qc_h - qc_csa) - qc_j - qc_j1, 0.to_d].max
```

Does not do the trick? Comparing with float instead?

Reply:
tried that of course... 
<img width="657" alt="image" src="https://github.com/Humi-HR/taxman/assets/134647356/1ca2f0b4-f748-41fd-bf0d-f5a8fe708d02">

i don't get this.  works fine in console

Reply:
oh wait i know why -- first value is NaN